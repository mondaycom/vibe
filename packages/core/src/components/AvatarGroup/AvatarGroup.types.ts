import { CounterColor } from "../Counter/Counter.types";

export type AvatarGroupCounterVisualProps = {
  color?: Extract<CounterColor, "light" | "dark">;
  count?: number;
  prefix?: string;
  maxDigits?: number;
  ariaLabelItemsName?: string;
  noAnimation?: boolean;
  tabIndex?: number;
  /**
   * Relevant only for when AvatarGroup contains a clickable avatar
   */
  dialogContainerSelector?: string;
};
