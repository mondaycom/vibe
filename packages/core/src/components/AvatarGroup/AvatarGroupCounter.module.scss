@use "sass:math";
@import "~monday-ui-style/dist/mixins";
@import "../../components/Avatar/AvatarConstants";

.counterContainer {
  background: var(--ui-background-color);
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: calc(-1 * var(--spacing-small));
  z-index: 1;
  border: 1px solid var(--primary-background-color);

  &:focus-visible,
  &.focus-visible {
    outline: none;
    @include focus-style(50%);
  }

  @mixin counterSize($size) {
    height: $size;
    min-width: $size;
    border-radius: math.div($size, 2);
  }
  &.xs {
    @include counterSize($avatar-size-xs);
    padding: 0 1px;
  }
  &.small {
    @include counterSize($avatar-size-small);
    padding: 0 1px;
  }
  &.medium {
    @include counterSize($avatar-size-medium);
    padding: 0 2px;
  }
  &.large {
    @include counterSize($avatar-size-large);
    padding: 0 var(--spacing-xs);
  }

  &.primary {
    color: var(--text-color-on-primary);
    background-color: var(--primary-color);
  }
  &.dark {
    color: var(--text-color-on-inverted);
    background-color: var(--inverted-color-background);
  }
  &.negative {
    color: var(--text-color-on-primary);
    background-color: var(--negative-color);
  }
  &.light {
    color: var(--primary-text-color);
    background-color: var(--ui-background-color);
  }

  .disabled,
  &.disabled {
    background-color: var(--disabled-background-color);
    color: var(--disabled-text-color);
  }
}

.menu {
  max-height: 300px;
  overflow-y: auto;
  overflow-x: hidden;
  @include scroller();
}
