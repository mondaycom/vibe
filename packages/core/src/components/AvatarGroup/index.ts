export { default as AvatarGroup, type AvatarGroupProps } from "./AvatarGroup";
export * from "./AvatarGroup.types";
