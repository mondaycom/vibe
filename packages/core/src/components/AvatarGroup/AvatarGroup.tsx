import React, { ReactElement, useMemo } from "react";
import cx from "classnames";
import { AvatarProps } from "../Avatar/Avatar";
import AvatarGroupCounter from "./AvatarGroupCounter";
import VibeComponentProps from "../../types/VibeComponentProps";
import { AvatarSize, AvatarType } from "../Avatar/Avatar.types";
import { avatarOnClick } from "./AvatarGroupHelper";
import { TooltipProps } from "../Tooltip/Tooltip";
import styles from "./AvatarGroup.module.scss";
import { AvatarGroupCounterVisualProps } from "./AvatarGroup.types";

export interface AvatarGroupProps extends VibeComponentProps {
  /**
   * Class name applied to each avatar.
   */
  avatarClassName?: string;
  /**
   * The avatars displayed in the group.
   */
  children?: ReactElement<AvatarProps> | ReactElement<AvatarProps>[];
  /**
   * The size of the avatars in the group.
   */
  size?: AvatarSize;
  /**
   * The type of the avatars in the group.
   */
  type?: AvatarType;
  /**
   * The maximum number of avatars displayed before showing a counter.
   */
  max?: number;
  /**
   * Props for customizing the counter display.
   */
  counterProps?: AvatarGroupCounterVisualProps;
  /**
   * Props passed to the Tooltip component. See full options in the [Tooltip documentation](https://vibe.monday.com/?path=/docs/components-tooltip--docs).
   */
  counterTooltipCustomProps?: Partial<TooltipProps>;
  /**
   * If true, the counter tooltip uses a virtualized list for performance optimization.
   */
  counterTooltipIsVirtualizedList?: boolean;
  /**
   * If true, the component is disabled and non-interactive.
   */
  disabled?: boolean;
}

const AvatarGroup: React.FC<AvatarGroupProps> = ({
  className,
  avatarClassName,
  id,
  children,
  size,
  type,
  max = 5,
  counterProps,
  counterTooltipCustomProps,
  counterTooltipIsVirtualizedList = false,
  disabled
}) => {
  const { displayAvatars, counterTooltipAvatars } = useMemo(() => {
    if (!children) {
      return {};
    }
    const childrenArray = Array.isArray(children) ? children : [children];
    return {
      displayAvatars: childrenArray.slice(0, max).map((avatar, index) => {
        return React.cloneElement(avatar, {
          key: index,
          ...avatar?.props,
          size: size || avatar?.props?.size,
          type: type || avatar?.props?.type,
          className: cx(styles.avatarContainer, avatarClassName),
          onClick: (event: React.MouseEvent | React.KeyboardEvent) => avatarOnClick(event, avatar.props),
          disabled
        });
      }),
      counterTooltipAvatars: childrenArray.slice(max)
    };
  }, [avatarClassName, children, disabled, max, size, type]);

  if (!children) {
    return null;
  }

  return (
    <div className={cx(styles.avatarGroupContainer, className)} id={id}>
      {displayAvatars}
      <AvatarGroupCounter
        counterTooltipAvatars={counterTooltipAvatars}
        counterProps={counterProps}
        counterTooltipCustomProps={counterTooltipCustomProps}
        counterTooltipIsVirtualizedList={counterTooltipIsVirtualizedList}
        size={size}
        type={type}
        disabled={disabled}
      />
    </div>
  );
};

export default AvatarGroup;
