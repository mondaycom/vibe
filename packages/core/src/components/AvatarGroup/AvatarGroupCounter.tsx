import React, { ReactElement, useCallback, useRef } from "react";
import cx from "classnames";
import { AvatarProps } from "../Avatar/Avatar";
import Counter from "../Counter/Counter";
import MenuButton from "../MenuButton/MenuButton";
import Menu from "../Menu/Menu/Menu";
import AvatarMenuItem from "../Menu/MenuItem/AvatarMenuItem";
import AvatarGroupCounterTooltipContainer from "./AvatarGroupCounterTooltipContainer";
import VibeComponentProps from "../../types/VibeComponentProps";
import { AvatarSize, AvatarType } from "../Avatar/Avatar.types";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import { AvatarGroupCounterVisualProps } from "./AvatarGroup.types";
import { TooltipProps } from "../Tooltip/Tooltip";
import { avatarOnClick } from "./AvatarGroupHelper";
import styles from "./AvatarGroupCounter.module.scss";
import { AVATAR_GROUP_COUNTER_AVATAR_SIZE } from "./AvatarGroupConstants";

export interface AvatarGroupCounterProps extends VibeComponentProps {
  /**
   * The list of avatars hidden behind the counter.
   */
  counterTooltipAvatars?: ReactElement<AvatarProps>[];
  /**
   * Props for customizing the counter appearance.
   */
  counterProps?: AvatarGroupCounterVisualProps;
  /**
   * Props for customizing the counter tooltip.
   */
  counterTooltipCustomProps?: Partial<TooltipProps>;
  /**
   * If true, the counter tooltip uses a virtualized list for performance optimization.
   */
  counterTooltipIsVirtualizedList?: boolean;
  /**
   * The size of the counter.
   */
  size?: AvatarSize;
  /**
   * The type of the avatars in the counter.
   */
  type?: AvatarType;
  /**
   * The label of the counter for accessibility.
   */
  counterAriaLabel?: string;
  /**
   * If true, the counter is disabled and non-interactive.
   */
  disabled?: boolean;
}

const AvatarGroupCounter: React.FC<AvatarGroupCounterProps> = ({
  counterTooltipAvatars = [],
  counterProps,
  counterTooltipCustomProps,
  counterTooltipIsVirtualizedList = false,
  size = "medium",
  type,
  counterAriaLabel,
  disabled
}: AvatarGroupCounterProps) => {
  const {
    color: counterColor = "light",
    count: counterValue = counterTooltipAvatars.length,
    prefix: counterPrefix = "+",
    maxDigits: counterMaxDigits = 3,
    ariaLabelItemsName: counterAriaLabelItemsName = "items",
    noAnimation,
    dialogContainerSelector,
    tabIndex: counterTabIndex = 0
  } = counterProps || {};

  const counterSizeStyle = getStyle(styles, size?.toString());
  const counterColorStyle = styles[counterColor];

  const focusPrevPlaceholderRef = useRef(null);
  const focusNextPlaceholderRef = useRef(null);
  const counterContainerRef = useRef(null);
  const counterComponent = useCallback(() => {
    return (
      <Counter
        color={counterColor}
        count={counterValue}
        prefix={counterPrefix}
        maxDigits={counterMaxDigits}
        ariaLabel={counterAriaLabel ? counterAriaLabel : `Tab for more ${counterAriaLabelItemsName}`}
        noAnimation={noAnimation}
        counterClassName={cx({ [styles.disabled]: disabled })}
        size={size === "xs" ? "xs" : undefined}
      />
    );
  }, [
    counterAriaLabel,
    counterAriaLabelItemsName,
    counterColor,
    counterMaxDigits,
    counterPrefix,
    counterValue,
    disabled,
    noAnimation
  ]);

  if (!counterTooltipAvatars.length && !counterValue) {
    return null;
  }

  if (disabled) {
    return (
      <div
        ref={counterContainerRef}
        className={cx(styles.counterContainer, styles.disabled, counterSizeStyle, counterColorStyle)}
      >
        {counterComponent()}
      </div>
    );
  }

  const areAvatarsClickable = counterTooltipAvatars.some(a => a.props?.onClick);
  if (areAvatarsClickable) {
    return (
      <MenuButton
        component={counterComponent}
        zIndex={1}
        className={cx(styles.counterContainer, counterSizeStyle, counterColorStyle)}
        ariaLabel={counterAriaLabel ? counterAriaLabel : `${counterValue} additional ${counterAriaLabelItemsName}`}
        dialogContainerSelector={dialogContainerSelector}
      >
        <Menu id="menu" size={Menu.sizes.MEDIUM} className={styles.menu} focusItemIndexOnMount={0}>
          {counterTooltipAvatars.map((avatar, index) => {
            return (
              // eslint-disable-next-line react/jsx-key
              <AvatarMenuItem
                menuItemProps={{
                  key: avatar.props?.id || String(index),
                  title: (avatar.props?.tooltipProps?.content as string) || avatar?.props?.ariaLabel,
                  onClick: (event: React.MouseEvent | React.KeyboardEvent) => avatarOnClick(event, avatar.props)
                }}
                avatarProps={{
                  ...avatar.props,
                  customSize: AVATAR_GROUP_COUNTER_AVATAR_SIZE,
                  ariaLabel: "",
                  tabIndex: -1
                }}
              />
            );
          })}
        </Menu>
      </MenuButton>
    );
  }

  return (
    <AvatarGroupCounterTooltipContainer
      focusPrevPlaceholderRef={focusPrevPlaceholderRef}
      focusNextPlaceholderRef={focusNextPlaceholderRef}
      counterContainerRef={counterContainerRef}
      avatars={counterTooltipAvatars}
      counterTooltipCustomProps={counterTooltipCustomProps}
      counterTooltipIsVirtualizedList={counterTooltipIsVirtualizedList}
      type={type}
    >
      <div tabIndex={-1} ref={focusPrevPlaceholderRef} />
      {/* eslint-disable jsx-a11y/no-noninteractive-tabindex */}
      <div
        tabIndex={counterTabIndex}
        className={cx(styles.counterContainer, counterSizeStyle, counterColorStyle)}
        ref={counterContainerRef}
      >
        {/* eslint-enable jsx-a11y/no-noninteractive-tabindex */}
        {counterComponent()}
        <div tabIndex={-1} ref={focusNextPlaceholderRef} />
      </div>
    </AvatarGroupCounterTooltipContainer>
  );
};

export default AvatarGroupCounter;
