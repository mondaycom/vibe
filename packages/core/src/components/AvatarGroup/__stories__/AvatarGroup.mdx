import { Meta } from "@storybook/blocks";
import { UsageGuidelines } from "vibe-storybook-components";
import { AVATAR, COUNTER, BADGE } from "../../../storybook/components/related-components/component-description-map";
import * as AvatarGroupStories from "./AvatarGroup.stories";

<Meta of={AvatarGroupStories} />

# AvatarGroup

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Use this component if you need to stack avatars as a group.

<Canvas of={AvatarGroupStories.Overview} />

### Import

```js
import { AvatarGroup } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    <>
      Use <StorybookLink page="Components/Tooltip">tooltip</StorybookLink> component while hovering on the counter when
      you need only to display the content
    </>,
    <>
      If clickable and navigable list is required on counter, use{" "}
      <StorybookLink page="Components/Menu/Menu">Menu</StorybookLink> component
    </>
  ]}
/>

## Variants

### Size

Avatar Group appears in 4 sizes: XS, Small, Medium, and Large.

<Canvas of={AvatarGroupStories.Size} />

### Color variants

You can use Light or Dark counter color to maintain visual hierarchy.

<Canvas of={AvatarGroupStories.ColorVariants} />

### Clickable vs. Hover

If avatars are clickable, they will be displayed via <StorybookLink page="Components/Menu/Menu">Menu</StorybookLink> and user will be able to navigate each additional item.
Otherwise, avatars will be displayed in a Tooltip with no item's navigation.

<Canvas of={AvatarGroupStories.HoverVsClickable} />

### Disabled

Use when the avatar group is inactive in the specific context.

<Canvas of={AvatarGroupStories.Disabled} />

### Max avatars shown

Choose the ammount of avatars you want to show

<Canvas of={AvatarGroupStories.MaxAvatarsToDisplay} />

### Custom counter

You can pass `counterProps` to specify counter params.

<Canvas of={AvatarGroupStories.CustomCounter} />

### Grid tooltip

When tooltip text for additional avatars is not passed, extra avatars will be displayed in a grid mode.

<Canvas of={AvatarGroupStories.GridTooltip} />

### Virtualized list

Should be used only to display large amount of avatars in default counter tooltip

<Canvas of={AvatarGroupStories.VirtualizedList} />

### Counter custom tooltip content

Counter tooltip props can be specified in order to render tooltip with custom content.

<Canvas of={AvatarGroupStories.CounterCustomTooltipContent} />

## Use cases and examples

### Last seen users

<Canvas of={AvatarGroupStories.LastSeenUsers} />

### Displaying teams

<Canvas of={AvatarGroupStories.DisplayingTeams} />

## Related components

<RelatedComponents componentsNames={[AVATAR, BADGE, COUNTER]} />
