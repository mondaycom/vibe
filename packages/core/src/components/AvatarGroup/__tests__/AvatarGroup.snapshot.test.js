import React from "react";
import renderer from "react-test-renderer";
import AvatarGroup from "../AvatarGroup";
import Avatar from "../../Avatar/Avatar";

// Component depends on Avatar, Counter and Tooltip components
describe("AvatarGroup renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<AvatarGroup />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with dummy class name", () => {
    const tree = renderer
      .create(
        <AvatarGroup className="dummy-class-name">
          <Avatar text="P1" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with counter", () => {
    const tree = renderer
      .create(
        <AvatarGroup max={1}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly without counter", () => {
    const tree = renderer
      .create(
        <AvatarGroup max={2}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom counter value", () => {
    const tree = renderer
      .create(
        <AvatarGroup max={2} counterValue={13}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with large size", () => {
    const tree = renderer
      .create(
        <AvatarGroup size="large" max={1}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly with xs size", () => {
    const tree = renderer
      .create(
        <AvatarGroup size="xs" max={1}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with counter ariaLabel default-tooltip", () => {
    const tree = renderer
      .create(
        <AvatarGroup max={1}>
          <Avatar text="P1" ariaLabel="Person 1" />
          <Avatar text="P2" ariaLabel="Person 2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with counter default-tooltip", () => {
    const tree = renderer
      .create(
        <AvatarGroup max={1}>
          <Avatar text="P1" tooltipProps={{ content: "Person 1" }} />
          <Avatar text="P2" tooltipProps={{ content: "Person 2" }} />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom counter tooltip", () => {
    const tree = renderer
      .create(
        <AvatarGroup counterTooltipCustomProps={{ content: "Custom tooltip content" }} max={1}>
          <Avatar text="P1" />
          <Avatar text="P2" />
        </AvatarGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
