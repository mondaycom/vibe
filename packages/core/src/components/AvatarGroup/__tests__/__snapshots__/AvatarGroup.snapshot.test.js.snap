// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup renders correctly renders correctly with counter 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer medium light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeLarge kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with counter ariaLabel default-tooltip 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          aria-label="Person 1"
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer medium light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeLarge kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with counter default-tooltip 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer medium light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeLarge kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with custom counter tooltip 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer medium light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeLarge kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with custom counter value 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P2
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with dummy class name 1`] = `
<div
  className="avatarGroupContainer dummy-class-name"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with large size 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer large light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeLarge kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly with xs size 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar xs avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextXs"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className=""
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={-1}
  />
  <div
    className="counterContainer xs light"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-label="Tab for more items 1"
      aria-labelledby=""
      onMouseDown={[Function]}
    >
      <div
        aria-label="1"
        className="counter sizeXs kindFill colorLight"
      >
        <span
          data-testid="counter"
          id="counter"
        >
          +1
        </span>
      </div>
    </span>
    <div
      tabIndex={-1}
    />
  </div>
</div>
`;

exports[`AvatarGroup renders correctly renders correctly without counter 1`] = `
<div
  className="avatarGroupContainer"
>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P1
        </span>
      </div>
    </div>
  </div>
  <div
    className="avatar large avatarContainer"
    data-testid="avatar"
    style={{}}
  >
    <div
      className="clickable clickableWrapper disableTextSelection"
      data-testid="clickable"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        aria-hidden={false}
        className="circle circleText"
        style={
          {
            "backgroundColor": "var(--color-chili-blue)",
          }
        }
        tabIndex={0}
      >
        <span
          className="contentText contentTextLarge"
          data-testid="avatar-content"
        >
          P2
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarGroup renders correctly with empty props 1`] = `null`;
